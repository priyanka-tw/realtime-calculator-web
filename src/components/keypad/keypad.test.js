import React from 'react';
import {act, fireEvent, render} from '@testing-library/react';
import 'jest-styled-components';
import Keypad from "./keypad";

describe('Keypad', () => {
    const mockFn = jest.fn();

    it('should render the component with default prop', () => {
        const {container} = render(<Keypad onClick={mockFn}/>);

        expect(container).toMatchSnapshot();
    });

    it('should perform the given action on click of a button', async () => {
        const {getByTestId} = render(<Keypad onClick={mockFn}/>);

        await act(async () => {
            await fireEvent.click(getByTestId('calc-button-1'));
        });

        expect(mockFn).toHaveBeenCalledTimes(1);
        expect(mockFn).toHaveBeenCalledWith("1");
    });

});
