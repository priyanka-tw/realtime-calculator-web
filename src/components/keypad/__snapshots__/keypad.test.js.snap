// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Keypad should render the component with default prop 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c2 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
}

.c3 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  color: #ffc107;
}

.c4 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  min-width: 120px;
}

.c5 {
  font-size: 16pt;
  font-family: verdana;
  min-width: 60px;
  height: 60px;
  outline: none;
  background-color: black;
  color: white;
  cursor: pointer;
  border: 1px solid #80808054;
  background-color: #ffc107;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <button
        class="c2"
        data-testid="digit-button-AC"
        name="AC"
        type="button"
      >
        AC
      </button>
      <button
        class="c2"
        data-testid="digit-button-C"
        name="C"
        type="button"
      >
        C
      </button>
      <button
        class="c2"
        data-testid="digit-button-%"
        name="%"
        type="button"
      >
        %
      </button>
      <button
        class="c3"
        data-testid="digit-button-/"
        name="/"
        type="button"
      >
        /
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        class="c2"
        data-testid="digit-button-7"
        name="7"
        type="button"
      >
        7
      </button>
      <button
        class="c2"
        data-testid="digit-button-8"
        name="8"
        type="button"
      >
        8
      </button>
      <button
        class="c2"
        data-testid="digit-button-9"
        name="9"
        type="button"
      >
        9
      </button>
      <button
        class="c3"
        data-testid="digit-button-*"
        name="*"
        type="button"
      >
        *
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        class="c2"
        data-testid="digit-button-4"
        name="4"
        type="button"
      >
        4
      </button>
      <button
        class="c2"
        data-testid="digit-button-5"
        name="5"
        type="button"
      >
        5
      </button>
      <button
        class="c2"
        data-testid="digit-button-6"
        name="6"
        type="button"
      >
        6
      </button>
      <button
        class="c3"
        data-testid="digit-button--"
        name="-"
        type="button"
      >
        -
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        class="c2"
        data-testid="digit-button-1"
        name="1"
        type="button"
      >
        1
      </button>
      <button
        class="c2"
        data-testid="digit-button-2"
        name="2"
        type="button"
      >
        2
      </button>
      <button
        class="c2"
        data-testid="digit-button-3"
        name="3"
        type="button"
      >
        3
      </button>
      <button
        class="c3"
        data-testid="digit-button-+"
        name="+"
        type="button"
      >
        +
      </button>
    </div>
    <div
      class="c1"
    >
      <button
        class="c4"
        data-testid="digit-button-0"
        name="0"
        type="button"
      >
        0
      </button>
      <button
        class="c2"
        data-testid="digit-button-."
        name="."
        type="button"
      >
        .
      </button>
      <button
        class="c5"
        data-testid="digit-button-="
        name="="
        type="button"
      >
        =
      </button>
    </div>
  </div>
</div>
`;
